### PR TITLE
Feat/error handling edit profile page

### PR DESCRIPTION
**Pull request by shan-226**

### CODE REVIEW

File: lib/pages/edit_profile_page.dart [hunk 1]
```diff
@@ -245,20 +280,20 @@ class  EditProfileFormState  extends State<EditProfileForm> {
       final FirebaseAuth auth = FirebaseAuth.instance;
       final currentUser = auth.currentUser;
       if (currentUser != null) {
-        if (username != "") {
+        if (username.isNotEmpty) {
           final data = { "username.profile_name" :username};
           await db.collection("profile_data").doc(currentUser.uid).update(data);
         }
-        if(firstname != "")
+        if(firstname.isEmpty)
```
Comment (MonikaTheiss) [inline lib/pages/edit_profile_page.dart:287]: @shan-226 I believe this should be isNotEmpty?

File: lib/pages/edit_profile_page.dart [hunk 2]
```diff
@@ -268,10 +303,11 @@ class  EditProfileFormState  extends State<EditProfileForm> {
       }
     }catch (e)
     {
+      throw Exception('Error updating profile: $e');
```
Comment (MonikaTheiss) [inline lib/pages/edit_profile_page.dart:306]: Im just wondering, should we let the user know that there was an error when they tried to update their profile? Let me know what you think?

**Review by MonikaTheiss — CHANGES_REQUESTED** [review]

File: lib/pages/edit_profile_page.dart (hunk 2 above)
Comment (pieryvdl) [reply lib/pages/edit_profile_page.dart:306]: I think we should.

File: lib/pages/edit_profile_page.dart (hunk 1 above)
Comment (shan-226) [reply lib/pages/edit_profile_page.dart:287]: Oops, thank you for spotting this. Fixed.

**Review by pieryvdl — COMMENTED** [review]
Just waiting for Monika's feedback to be addressed.

File: lib/pages/edit_profile_page.dart (hunk 2 above)
Comment (shan-226) [reply lib/pages/edit_profile_page.dart:306]: I honestly don't know. Some users might like to see it instead of wondering why it didn't work?

**Review by pieryvdl — APPROVED** [review]
Seems to work.

**Review by MonikaTheiss — APPROVED** [review]

**Pull request merged**